tests/management/test_resource_manager.py

```diff
@@ -2036,3 +2039,25 @@ def test_locking_resource_with_group_none(self):
                          % (self.NO_GROUP_RESOURCE, resource.name))
 
         self.get_resource(self.NO_GROUP_RESOURCE, owner="")
+
+
+class ClientWebsocketTests(TestCase):
+    """Tests for client behavior."""
+
+    TEST_ADDRESS = 'localhost'
+
+    @mock.patch('websocket._core.handshake')
+    @mock.patch('websocket._core.connect')
+    @mock.patch('rotest.management.client.websocket_client.'
+                'PingingWebsocket.handle_disconnection')
+    def test_server_disconnection(self, mock_callback, mock_connect, _):
+        """Check that 'close_session' is called at disconnect."""
+        client = PingingWebsocket(ping_interval=0.2)
+        mock_socket = mock.MagicMock()
+        mock_connect.return_value = mock_socket, self.TEST_ADDRESS
+        client.connect(self.TEST_ADDRESS)
+        time.sleep(0.6)
```

Comment:
Use waiting library or something,
It's too flaky relying on the current performance of the code